ludwig/schema/encoders/text_encoders.py

```diff
@@ -2960,6 +2963,12 @@ def module_name():
 class AutoTransformerConfig(HFEncoderConfig):
     """This dataclass configures the schema used for an AutoTransformer encoder."""
 
+    def __post_init__(self):
+        if self.pretrained_model_name_or_path is None:
+            raise ConfigValidationError(
+                "`pretained_model_name_or_path` must be specified for encoder: `auto_transformer`."
+            )
```

Comment:
Interesting.

This does seems potentially cleaner than as an aux check as with the aux check you would need to first parse through the config to find text features that use the "auto_transformer" encoder.

Potentially this would be the preferred convention for validation checks for configs with incompatible interdependent parameter values within the same section of the config, like `check_training_runway`.

Does `base.py::ModelConfig`'s `__post_init__` get called after this one?

Reply:
Yes, that's right. I like this approach for two reasons:

1. As you said, it means we don't need to parse through the config to hunt for these features, which makes things better encapsulated.
2. If in the future we want users to interact with the config directly by instantiating these schema components, then being able to perform validation at the component level (rather than just the config level) will make it more user-friendly.